invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -24,11 +25,53 @@
 #
 # Helpers
 #
+
+def get_form_pids_config():
+    """Prepare configuration for the pids field."""
+    service = current_rdm_records.records_service
+    pids_providers = []
+    for scheme, providers in service.config.pids_providers.items():
+        can_be_managed = False
+        can_be_unmanaged = False
+        provider_enabled = False
+        for name, provider_attrs in providers.items():
+            is_enabled = provider_attrs.get("enabled", True)
+            if not provider_enabled and is_enabled:
+                provider_enabled = True
+
+            if provider_attrs["system_managed"]:
+                can_be_managed = True
+            else:
+                can_be_unmanaged = True
+
+        # all providers disabled for this scheme
+        if not provider_enabled:
+            continue
+
+        u_scheme = scheme.upper()
+
+        pids_provider = {
+            "scheme": scheme,
+            "pid_label": u_scheme,
+            "pid_placeholder": "10.1234/rdm.123456",
+            "can_be_managed": can_be_managed,
+            "can_be_unmanaged": can_be_unmanaged,
+            "btn_label_get_pid": f"Get a {u_scheme} now!",
+            "managed_help_text": f"Reserve a {u_scheme} or leave this " \
```

Comment:
minor: This should be "managed" and "required", right? i.e. even if a pid type is managed if there is no required provider, it wont automatically get a pid.

Reply:
For the time being, I did not add a `required` to the UI: if the user will not insert/reserve anything, and the backend has the managed required, it will automatically create a DOI.